lib/app_coordinator.dart

```diff
@@ -0,0 +1,85 @@
+import 'package:farmsmart_flutter/data/repositories/repository_provider.dart';
+import 'package:farmsmart_flutter/ui/home.dart';
+import 'package:flutter/material.dart';
+
+import 'deep_link_helper.dart';
+import 'utils/shared_preferences_helper.dart';
+
+class AppCoordinator extends StatefulWidget {
+  @override
+  _AppCoordinatorState createState() => _AppCoordinatorState();
+}
+
+class _AppCoordinatorState extends State<AppCoordinator> {
+  RepositoryProvider repositoryProvider;
+
+  @override
+  void initState() {
+    super.initState();
+    DeepLinkHelper(deepLinks: _deepLinks()).init();
+
+    repositoryProvider = RepositoryProvider(context: context);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    repositoryProvider.init();
+
+    //TODO Replace by appEntryManagement
+    return Home(repositoryProvider: repositoryProvider,);
+  }
+
+  //TODO Should be implemented when the landing page is merged
+  FutureBuilder<bool> appEntryManagement(){
+    return FutureBuilder<bool>(
+      future: SharedPreferencesHelper.getIsFirstLaunch(),
+      builder: (BuildContext context, AsyncSnapshot<bool> snapshot){
+        if(snapshot.hasData){
+          SharedPreferencesHelper.setIsFirstLaunch(false);
+
+          //TODO Replace Text widget by LandingPage if snapshot.data is true
+          return snapshot.data ? Text('FIRST LAUNCH') : Home(repositoryProvider: repositoryProvider,);
+        }else{
+          return Container();
+        }
+      },
+    );
+  }
+
+  /*
+    TODO Implement the right action and the right deep link parameter,
+    This is just an example
+   */
+  List<DeepLink> _deepLinks() {
+    return [
+      DeepLink(
+        deepLinkParameter: 'articleId',
+        action: (deepLinkResult) {
+          Navigator.of(context).push(
+            MaterialPageRoute(
+              builder: (context) => Center(
+                child: Container(
+                    child: Text(
+                        'Opened article dynamic link with $deepLinkResult')),
+              ),
+            ),
+          );
+        },
+      ),
+      DeepLink(
```

Comment:
Can we move that to a method that creates the deeplink?

Reply:
### Read the TODO above :D

Reply:
Monday.. <image src="https://user-images.githubusercontent.com/15966505/62059613-e65d8980-b223-11e9-8adb-ac7e98c05663.png" width="10%">